packages/object-map/index.test.ts

```diff
@@ -0,0 +1,35 @@
+import map from "./index";
+
+// OK
+map({ a: "cow", b: "sheep" }, (key, value) => key);
+map({ a: "cow", b: "sheep" }, (key, value) => value);
+map({ a: "cow", b: null }, (key, value) => value);
+map({ a: "cow", b: "sheep" }, (key, value) => key + value);
+map({ a: "cow", b: null }, (key, value) => key + value);
+map([1, 2, 3], (key, value) => key);
+map([1, 2, 3], (key, value) => value);
+map([1, 2, 3], (key, value) => key + value);
+
+// Not OK
+//@ts-expect-error
+map();
+//@ts-expect-error
+map("hello");
+//@ts-expect-error
+map(1);
+//@ts-expect-error
+map(true);
+//@ts-expect-error
+map("hello", () => null);
+//@ts-expect-error
+map(true, () => null);
+//@ts-expect-error
+map(1, () => null);
+//@ts-expect-error
+map({ a: true, b: false }, (key, value) => 2 * value);
```

Comment:
First 3 are invalid as the argument is not an object and 4th would be invalid as value is an boolean and we can not do mathematical operations on them.

Yes, one can still typecast key and value to do desirable operations but this would give them error on compile time if they are not typecasting.

Reply:
Sorry I was not clear, I meant starting from line 29

Reply:
So regarding tests on lines 29, 31, 33 and 35...
I guess I don't know enough about how typescript works but it seems wrong to me to invalidate a statement that will resolve to a legitimate value via coercion.